### DISCUSSION TITLE
[feature request] Handle key states and update icon based on state

### DISCUSSION THREAD

**Discussion by prankousky** [streamdeck-linux-gui/streamdeck-linux-gui#26] [Ideas / Feature Requests]
Hi,

this project has been recommended to me via github issues in another (stale) project. I'd like to suggest something that would really make this so much more usable to me: icon based on state.

Let's say we have an icon that toggles a light. This would allow us to set these properties to the icon:

* current state (`on` / `off`)
* icon state `on`
* icon state `off`
* icon color `on`
* icon color `off`
 
Icon color would only apply (or make sense) if the icon was monochrome. So I'll just assume it is.

The state would have to be defined either via command, for example, `curl https://my-server.local/states/light00`, which would, in this case, return `on` or `off` - or it could be determined by reading a file, which gets modified by a third-party script, for example `/tmp/states_light00` (which would only contain `0` / `1`, or `true` / `false`, or `on` / `off`).

Icon color could be defined just as above, but additionally just set via GUI. For example, `on = orange` or `on = #f27013` and `off = white / black` or `off = #FFFFFF / #000000`.

Finally, icon state could be handled the same way as icon color: either load the icon path via command, read from file, or set the values for `on` and `off` via GUI.

I like to control `Home Assistant` via Streamdeck. This is possible via https://github.com/basnijholt/home-assistant-streamdeck-yaml . However, this **only** let's me control Home Assistant. If this project would support icon color and state icons, I could control Home Assistant **and** my desktop simultaneously. 

The Home Assistant community is pretty large, so I could assume a lot of people having the need for something like this.

Just an idea :)

**Comment by Danie10** [comment]
It may be related also to issue #5, although the description of Dynamic Buttons does not really describe it completely.

**Reply by coolapso** [reply]
believe both features are fundamentally different even though they might have an overlap. 

I'd say this one is the basic foundation for #5, for this the application needs to handle key-press events and keep the state of the button presses somewhere.

There are more feature requests that could benefit / conflict with this such as: https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/discussions/17

**Comment by coolapso** [comment]
@prankousky I have edited the title of this topic to better reflect the feature as this is the core for many features being requested.

**Reply by StormyIceLeopard** [reply]
This could be a nice feature to help with OBS integration, similar to Windows Version.
Would help resolve this one https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/discussions/25

**Comment by neutropico** [comment]
It would be nice something like https://docs.xfce.org/panel-plugins/xfce4-genmon-plugin/start
